src/fhirtypes/ElementDefinition.ts

```diff
@@ -319,10 +319,20 @@ export class ElementDefinition {
   }
 
   validate(): ValidationError[] {
+    const errors: ValidationError[] = [];
     if (this.slicing) {
-      return this.validateSlicing(this.slicing);
+      if (this.sliceName) {
+        errors.push(
+          new ValidationError(
+            'An element with a slice name does not need to define its own slicing. Instead, define additional slicing discriminators on the base element.',
```

Comment:
I think it needs to be just a tad stronger:

```suggestion
            'An element with a slice name should not define its own slicing. Instead, define additional discriminators to the original slicing on the base element.',
```